pkg/dmapi/master_control.go

```diff
@@ -74,6 +75,28 @@ func (mc *defaultMasterControl) GetMasterClient(namespace Namespace, dcName stri
 	return mc.masterClients[key]
 }
 
+func (mc *defaultMasterControl) GetMasterPeerClient(namespace Namespace, dcName string, podName string, tlsEnabled bool) MasterClient {
+	mc.mutex.Lock()
+	defer mc.mutex.Unlock()
+
+	var tlsConfig *tls.Config
+	var err error
+	var scheme = "http"
+
+	if tlsEnabled {
+		scheme = "https"
+		tlsConfig, err = pdapi.GetTLSConfig(mc.kubeCli, pdapi.Namespace(namespace), dcName, util.ClusterClientTLSSecretName(dcName))
+		if err != nil {
+			klog.Errorf("Unable to get tls config for dm cluster %q, master client may not work: %v", dcName, err)
+			return &masterClient{url: MasterPeerClientURL(namespace, dcName, podName, scheme), httpClient: &http.Client{Timeout: DefaultTimeout}}
+		}
+
+		return NewMasterClient(MasterPeerClientURL(namespace, dcName, podName, scheme), DefaultTimeout, tlsConfig)
+	}
+
+	return &masterClient{url: MasterPeerClientURL(namespace, dcName, podName, scheme), httpClient: &http.Client{Timeout: DefaultTimeout}}
```

Comment:
No need to cache the client?

Reply:
The upgrade request should not happen frequently, so I think we needn't open a keep-alive HTTP session.

Reply:
But here the configuration still makes it a keep-alive connection.